src/App.tsx

```diff
@@ -15,36 +16,112 @@ export const goodsFromServer = [
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+}
+
+interface SortParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortBy.Alphabetically:
+          return good1.localeCompare(good2);
+
+        case SortBy.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortBy.Alphabetically,
+          })}
+          onClick={() => {
+            setSortField(SortBy.Alphabetically);
+          }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SortBy.Length,
+          })}
+          onClick={() => {
+            setSortField(SortBy.Length);
+          }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortGoods.map(good => (
+          <li data-cy="Good" key={good + sortGoods.indexOf(good)}>
```

Comment:
```suggestion
          <li data-cy="Good" key={good}>
```
- You can get the index in the iteration method itself as an argument, so `indexOf` doesn't make sense
- It's a bad idea to generate keys at the same time when you render them, it's the same as usage index as a key. In this case good itself as a key is enough